daemon/cluster/cluster.go

```diff
@@ -725,6 +730,24 @@ func (c *Cluster) UpdateService(serviceID string, version uint64, spec types.Ser
 		return err
 	}
 
+	if encodedAuth != "" {
+		serviceSpec.Task.Runtime.(*swarmapi.TaskSpec_Container).Container.PullOptions = &swarmapi.ContainerSpec_PullOptions{RegistryAuth: encodedAuth}
```

Comment:
This line needs to use `GetContainer()` (and check the result) as well.

For consistency, we might as well do this in `CreateService` too.